packages/react-admin-core/src/table/useTableQueryPaging.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from "react";
+
+export interface IPagingApi<T> {
+    current: T;
+    currentPage?: number;
+    changePage: (variables: T, page?: number) => void;
+}
+export function useTableQueryPaging<T>(init: T): IPagingApi<T> {
+    const [page, setPage] = React.useState(1);
```

Comment:
Why do we need the extra `page` state? In your examples the page is used as `useTableQueryPaging<number>(page)`.

Reply:
for (optionally) showing the current page number (page x of y) - which is possibly not part of the T (when using cursor based paging)

It is also supported to have no page numbers (if the API doesn't provide them)